spec/lib/dispatch-rider/logging/translator_spec.rb

```diff
@@ -0,0 +1,173 @@
+  let(:duration) { nil }
+  let(:reason) { nil }
+
+  describe '.translate' do
+    # subject(:result) do
+    #   described_class.translate(message: fs_message, kind: kind, exception: exception, duration: duration, reason: reason)
+    # end
+
+    context 'when the job is starting' do
+      subject(:result) do
+        described_class.translate(fs_message, kind)
+      end
+      let(:kind) { :start }
+      let(:expected_hash) do
+        {
+          phase: :start,
+          subject: 'sample_handler',
+          guid: DispatchRider::Debug::PUBLISHER_MESSAGE_GUID,
+          body: {
+            'key' => 'value',
+          },
+        }
+      end
+
+      example do
+        expect(result).to eq(expected_hash)
+      end
+    end
+
+    context 'when the job has succeeded' do
+      subject(:result) do
+        described_class.translate(fs_message, kind)
+      end
+      let(:kind) { :success }
+      let(:expected_hash) do
+        {
+          phase: :success,
+          subject: 'sample_handler',
+          guid: DispatchRider::Debug::PUBLISHER_MESSAGE_GUID,
+          body: {
+            'key' => 'value',
+          },
+        }
+      end
+
+      example do
+        expect(result).to eq(expected_hash)
+      end
+    end
+
+    context 'when the job has failed' do
+      subject(:result) do
+        described_class.translate(fs_message, kind, exception: exception)
+      end
+      let(:kind) { :fail }
+      let(:exception) do
+        begin
+          raise(ArgumentError, "Foo is not bar")
+        rescue Exception => exception
+          exception
+        end
+      end
+
+      let(:expected_hash) do
+        {
+          phase: :fail,
+          subject: 'sample_handler',
+          guid: DispatchRider::Debug::PUBLISHER_MESSAGE_GUID,
+          body: {
+            'key' => 'value',
+          },
+          exception: {
+            class: 'ArgumentError',
+            message: 'Foo is not bar',
+          }
+        }
+      end
+
+      example do
+        expect(result).to eq(expected_hash)
+      end
+    end
+
+    context 'when the job has completed' do
+      subject(:result) do
+        described_class.translate(fs_message, kind, duration: duration)
+      end
+      let(:kind) { :complete }
+      let(:duration) { 5 }
+      let(:expected_hash) do
+        {
+          phase: :complete,
+          subject: 'sample_handler',
+          guid: DispatchRider::Debug::PUBLISHER_MESSAGE_GUID,
+          body: {
+            'key' => 'value',
+          },
+          duration: 5,
+        }
+      end
+
+      example do
+        expect(result).to eq(expected_hash)
+      end
+    end
+
+    context 'when the error handler fails' do
+      subject(:result) do
+        described_class.translate(fs_message, kind, exception: exception)
+      end
+      let(:kind) { :error_handler_fail }
+      let(:exception) do
+        begin
+          raise(ArgumentError, "Foo is not bar")
+        rescue Exception => exception
```

Comment:
Avoid rescuing the `Exception` class. Perhaps you meant to rescue `StandardError`?